src/llfs/volume_multi_append.test.cpp

```diff
@@ -0,0 +1,197 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/volume.hpp>
+//
+#include <llfs/volume.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <llfs/testing/test_config.hpp>
+
+#include <llfs/storage_simulation.hpp>
+
+#include <random>
+
+namespace {
+
+using namespace llfs::int_types;
+using namespace llfs::constants;
+
+using llfs::StatusOr;
+
+TEST(VolumeMultiAppendTest, SimTest)
+{
+  const std::string nonterminal(197, 'a');
+  const std::string terminal(193, 'b');
+
+  llfs::testing::TestConfig test_config;
+
+  const u32 kNumSeeds = test_config.extra_testing() ? 100 * 1000 : 400;
+
+  for (u32 seed = 0; seed < kNumSeeds; ++seed) {
+    for (usize n_before_crash : {1, 2, 3, 5, 10, 100, 500, 1000, 2000}) {
+      LLFS_VLOG(1) << BATT_INSPECT(seed) << BATT_INSPECT(n_before_crash);
+      std::mt19937 rng{seed};
+
+      llfs::StorageSimulation sim{batt::StateMachineEntropySource{
+          /*entropy_fn=*/[&rng](usize min_value, usize max_value) -> usize {
+            std::uniform_int_distribution<usize> pick_value{min_value, max_value};
+            return pick_value(rng);
+          }}};
+
+      sim.run_main_task([&] {
+        //+++++++++++-+-+--+----- --- -- -  -  -   -
+        {
+          //----- --- -- -  -  -   -
+          // Recover the simulated Volume.
+          //
+          LLFS_VLOG(1) << "sim.get_volume()";
+
+          StatusOr<std::unique_ptr<llfs::Volume>> recovered_volume = sim.get_volume(
+              "TestVolume", /*slot_visitor_fn=*/
+              [](auto&&...) {
+                return batt::OkStatus();
+              },
+              /*root_log_capacity=*/64 * kKiB);
+
+          ASSERT_TRUE(recovered_volume.ok()) << recovered_volume.status();
```

Comment:
Generally in tests I prefer to use `ASSERT_...` rather than `BATT_CHECK` as it at least allows other tests to run, rather than crashing the test process.